### ISSUE TITLE
Difference in scale and inference on a large tiff remote image 

### ISSUE THREAD

**Issue by a13383622838** [SarahwXU/HiSup#38]
Thank you for your code. I would like to ask if HiSup can perform inference on large images, for example, if I train on images of 256×256 pixels, can it infer different sizes or even very large remote sensing images during inference?

**Comment by XJKunnn** [comment]
Hi, thanks for your interest in our work. 

HiSup is able to handle images with different resolutions during inference stage. However, for very large-scale images, directly feeding the original image into the model may result in poor polygon generation. To achieve the best performance, it is recommended to use crop-and-merge inference process for large images. You can find the script [here](https://github.com/SarahwXU/HiSup/blob/88bd55036c8c6614f1fb76c7c86eed0839c04f1b/tools/test_pipelines.py#L162).

**Comment by a13383622838** [comment]
Thank you for your reply, I successfully ran through the code to reason about a large remote sensing image and got its vector shp file (you can directly match the original image in ArcGIS), I want to try to use HiSup to extract the red tide (irregular shape) in the remote sensing image, as shown in the figure below, I found that some red tide has been omitted, although the sea water does not appear to be wrongly mentioned, but the white bright bar on the edge of the island is wrongly mentioned, what do you think is the cause?
Here are some of the things I do:
1. Modified the normalization parameters to match my own red tide dataset
2. Set the original picture resolution in the profile to 256×256
3. Sliding window sets the window size to 256 and the step size to 128
4. 100 epochs of training on 1300 pictures of red tides in png format of 256×256
5. Inference 13400×12000 tif (RGB)

<img width="395" alt="Image" src="https://github.com/user-attachments/assets/e86ca4b8-a782-4455-a32c-a464298f6ac5" />

<img width="419" alt="Image" src="https://github.com/user-attachments/assets/7078ad04-667c-486f-bb7b-20fc27ad13a7" />

**Comment by XJKunnn** [comment]
Hi, thanks for providing the details. You can first visualize and check the predictions of mask and junction separately to see if the issue is caused by mask prediction or junction prediction. In this case, thresholds for mask and junction prediction can be adjusted to get the best performance. If both of the mask and junction are predicted correctly, check the process of polygonization. You can also adjust the parameters in polygonization process to make HiSup best fit in your case. If you have any question, feel free to share your new investigation here.

**Comment by a13383622838** [comment]
> Hi, thanks for providing the details. You can first visualize and check the predictions of mask and junction separately to see if the issue is caused by mask prediction or junction prediction. In this case, thresholds for mask and junction prediction can be adjusted to get the best performance. If both of the mask and junction are predicted correctly, check the process of polygonization. You can also adjust the parameters in polygonization process to make HiSup best fit in your case. If you have any question, feel free to share your new investigation here.

Thank you for your answer. I also want to know which loss can indicate whether the training is overfitting.
